zipline/utils/events.py

```diff
@@ -436,23 +438,33 @@ def date_func(self, dt, env):
         raise NotImplementedError
 
     def calculate_start_and_end(self, dt, env):
-        next_trading_day = _coerce_datetime(
-            env.add_trading_days(
-                self.td_delta,
-                self.date_func(dt, env),
+        new_date = self.date_func(dt, env)
+        if not new_date:
+            return
+
+        try:
+            next_trading_day = _coerce_datetime(
+                env.add_trading_days(
+                    self.td_delta,
+                    new_date,
+                )
             )
-        )
+        except (NoFurtherDataError, TypeError):
+            return
```

Comment:
Instead of keeping this in sync with below, what do you think about making a starting sentinel value (like `None`) for `next_trading_day`?  Like elsewhere we have `while dt and ...`